### PR TITLE
 Add keyboard: Vault35 WKL universal

### PR DESCRIPTION
**Pull request by josh-l-wang**
## Description

This is a PR to add a PCB I've made for the Vault 35 WKL cases. This PCB covers multiple layout options (ortho, traditional row stagger, katana, alpha, treadstone, and uniform staggers).

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* n/a

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/jlw/vault35_wkl_universal/info.json [hunk 1]
```diff
@@ -0,0 +1,769 @@
+{
+    "manufacturer": "jlw",
+    "keyboard_name": "vault 35 WKL universal",
+    "maintainer": "jlw",
+    "bootloader": "stm32-dfu",
+    "bootmagic": {
+        "matrix": [0, 0]
+    },
```
Comment (waffle87) [inline keyboards/jlw/vault35_wkl_universal/info.json:8]:
```suggestion
```

File: keyboards/jlw/vault35_wkl_universal/info.json [hunk 2]
```diff
@@ -0,0 +1,769 @@
...
+            {"matrix": [3, 0], "x": 21, "y": 88, "flags": 4},
+            {"matrix": [3, 0], "x": 12, "y": 88, "flags": 4},
+            {"matrix": [3, 0], "x": 1, "y": 88, "flags": 4},
+            {"matrix": [3, 0], "x": 6, "y": 80, "flags": 4},
+            {"matrix": [3, 0], "x": 1, "y": 70, "flags": 4},
+            {"matrix": [2, 0], "x": 1, "y": 58, "flags": 4},
+            {"matrix": [2, 0], "x": 1, "y": 46, "flags": 4},
+            {"matrix": [1, 0], "x": 1, "y": 34, "flags": 4}
+        ],
+        "max_brightness": 255,
```
Comment (waffle87) [inline keyboards/jlw/vault35_wkl_universal/info.json:151]:
```suggestion
```

File: keyboards/jlw/vault35_wkl_universal/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    ),
+
+    [1] = LAYOUT_all(
+         _______,  _______,  _______,  _______,  _______,  _______, _______,  KC_7,     KC_8,     KC_9,    KC_0,
+         _______,  _______,  _______,  _______,  _______,  _______, _______,  KC_4,     KC_5,     KC_6,    _______,
+         _______,  _______,  _______,  _______,  _______,  QK_BOOT, _______,  KC_1,     KC_2,     KC_3,    _______,
+         _______,            _______,  _______,  _______,  _______, _______,  _______,                     _______
+    ),
+
+    [2] = LAYOUT_all(
```
Comment (waffle87) [inline keyboards/jlw/vault35_wkl_universal/keymaps/default/keymap.c:21]: This layer is not accessible.

File: keyboards/jlw/vault35_wkl_universal/mcuconf.h [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+#undef STM32_PWM_USE_TIM17
+#define STM32_PWM_USE_TIM17 TRUE
+#define STM32_TIM17_SUPPRESS_ISR
+
+#define WS2812_PWM_DRIVER PWMD17
+#define WS2812_PWM_CHANNEL 1
+#define WS2812_PWM_PAL_MODE 2
+#define WS2812_PWM_COMPLEMENTARY_OUTPUT
+#define WS2812_DMA_STREAM STM32_DMA1_STREAM1
+#define WS2812_DMA_CHANNEL 1
```
Comment (waffle87) [inline keyboards/jlw/vault35_wkl_universal/mcuconf.h:17]: These definitions should all be moved to `config.h` file.
```suggestion
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/jlw/vault35_wkl_universal/keymaps/default/keymap.c (hunk 3 above)
Comment (josh-l-wang) [reply keyboards/jlw/vault35_wkl_universal/keymaps/default/keymap.c:21]: added MO(2) in layter 1

File: keyboards/jlw/vault35_wkl_universal/info.json (hunk 1 above)
Comment (josh-l-wang) [reply keyboards/jlw/vault35_wkl_universal/info.json:8]: I am not understanding the suggestion here.

File: keyboards/jlw/vault35_wkl_universal/mcuconf.h (hunk 4 above)
Comment (josh-l-wang) [reply keyboards/jlw/vault35_wkl_universal/mcuconf.h:17]: Moved.

File: keyboards/jlw/vault35_wkl_universal/info.json (hunk 1 above)
Comment (waffle87) [reply keyboards/jlw/vault35_wkl_universal/info.json:8]: This is the default functionality. It does not need to be redefined.

File: keyboards/jlw/vault35_wkl_universal/keymaps/default/keymap.c (hunk 3 above)
Comment (waffle87) [reply keyboards/jlw/vault35_wkl_universal/keymaps/default/keymap.c:21]: Doesn't appear so?

File: keyboards/jlw/vault35_wkl_universal/mcuconf.h (hunk 4 above)
Comment (waffle87) [reply keyboards/jlw/vault35_wkl_universal/mcuconf.h:17]: These values are to be removed from this file as well.

File: keyboards/jlw/vault35_wkl_universal/info.json (hunk 2 above)
Comment (waffle87) [reply keyboards/jlw/vault35_wkl_universal/info.json:151]: Please do not mark things as resolved without addressing them. This is to be removed, as it is the default value and does not need to be redefined.

**Comment by josh-l-wang** [comment]
Sorry about that, I made the changes last night and committed but I guess I messed up the commit and did not commit the changed files

File: keyboards/jlw/vault35_wkl_universal/info.json (hunk 1 above)
Comment (josh-l-wang) [reply keyboards/jlw/vault35_wkl_universal/info.json]: I see. Corrected

File: keyboards/jlw/vault35_wkl_universal/keymaps/default/keymap.c (hunk 3 above)
Comment (josh-l-wang) [reply keyboards/jlw/vault35_wkl_universal/keymaps/default/keymap.c:21]: Should be corrected now https://github.com/qmk/qmk_firmware/pull/23519/commits/91812274b7b47d1a2a31c504df795d2eed5ec1fa

File: keyboards/jlw/vault35_wkl_universal/config.h [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#define WS2812_PWM_DRIVER PWMD17
+#define WS2812_PWM_CHANNEL 1
+#define WS2812_PWM_PAL_MODE 2
+#define WS2812_PWM_COMPLEMENTARY_OUTPUT
+#define WS2812_DMA_STREAM STM32_DMA1_STREAM1
+#define WS2812_DMA_CHANNEL 1
```
[the remark this answers is not in the document]
Comment (josh-l-wang) [reply keyboards/jlw/vault35_wkl_universal/config.h]: Switched to master branch

File: keyboards/jlw/vault35_wkl_universal/info.json
[the remark this answers is not in the document]
Comment (josh-l-wang) [reply keyboards/jlw/vault35_wkl_universal/info.json:1]: Switched to master branch. I misread the PR checklist. Thanks!

**Comment by josh-l-wang** [comment]
I changed the target branch to master. I was mistaken in my interpretation of the PR checklist in the docs.

**Comment by josh-l-wang** [comment]
Okay rebased onto `master` and picked out my commits.

**Review by drashna — APPROVED** [review]

**Pull request merged**